### PR TITLE
description configurable

### LINKED ISSUE #169: Module Fails if Description is not added.
Found a bug? Maybe our [Slack Community](https://slack.cloudposse.com) can help. 

[Slack Community](https://slack.cloudposse.com)

## Describe the Bug
expected "description" to not be an empty string, got "", when `module.this (cloudposse/terraform-null-label)` is not integrated with the code. 
Line number [120](https://github.com/cloudposse/terraform-aws-elasticache-redis/blob/a580353a2ad35da4db3c12dabbc1750c16c47eb2/main.tf#L120) of `main.tf`

## Expected Behavior
The build should pass without issues.

## Steps to Reproduce
1. Configure terraform module with the dependent configuration as per the example
2. do not include `module "this"` as per README.md, on your local
3. carry out terraform plan

## Screenshots
Log.
```
╷
│ Error: expected "description" to not be an empty string, got 
│ 
│   with aws_elasticache_replication_group.default[0],
│   on main.tf line 120, in resource "aws_elasticache_replication_group" "default":
│  120:   description                = module.this.id
│ 
╵
```

## Environment (please complete the following information):
```
Stack: Terraform
OS: Linux Arch
Terraform: > 0.13
```

### PR DESCRIPTION
**Pull request by pen-pal**
This PR is to make sure description of replication group is configurable.

## what
* The deployment fails if `module "this"` as per example in README.md is not defined in local
* 

## why
* Should be able to configure description on my own without specifying any new modules.

## references
* Resolves [bug](https://github.com/cloudposse/terraform-aws-elasticache-redis/issues/169)
* Use `closes #169 `, if this PR closes a GitHub issue `#169`

### CODE REVIEW

**Review by dotCipher — APPROVED** [review]

**Comment by dotCipher** [comment]
Looks like you have an syntax error from a deprecated usage @pen-pal 
```
Error: Invalid combination of arguments
  on ../../main.tf line 120, in resource "aws_elasticache_replication_group" "default":
        	            	 120:   description                = var.description == "" ? module.this.id : var.description
        	            	"description": one of `description,replication_group_description` must be
        	            	specified
}
        	Test:       	TestExamplesComplete
TestExamplesComplete [202](https://github.com/cloudposse/actions/runs/7031269325?check_suite_focus=true#step:9:203)2-06-23T20:24:23Z retry.go:91: terraform [destroy -auto-approve -input=false -var attributes=["10976"] -var-file fixtures.us-east-2.tfvars -lock=false]
TestExamplesComplete 2022-06-23T20:24:23Z logger.go:66: Running command terraform with args [destroy -auto-approve -input=false -var attributes=["10976"] -var-file fixtures.us-east-2.tfvars -lock=false]
TestExamplesComplete 2022-06-23T20:24:28Z logger.go:66: 
TestExamplesComplete 2022-06-23T20:24:28Z logger.go:66: Warning: Argument is deprecated
```

**Comment by pen-pal** [comment]
> 

how should I fix it ?

**Comment by jtgorny** [comment]
What is the status of this? Seeing same exact error.

**Comment by pen-pal** [comment]
i am waiting on @dotCipher to help out resolve the issue

File: main.tf [hunk 1]
```diff
@@ -117,7 +117,7 @@ resource "aws_elasticache_replication_group" "default" {
 
   auth_token                 = var.transit_encryption_enabled ? var.auth_token : null
   replication_group_id       = var.replication_group_id == "" ? module.this.id : var.replication_group_id
-  description                = module.this.id
+  description                = var.description == "" ? module.this.id : var.description
```
Comment (joe-niland) [inline main.tf:120]:
```suggestion
  description                = coalesce(var.description, module.this.id)
```
@pen-pal see if this helps

**Comment by pen-pal** [comment]
@adamcrews @florian0410 anything I can do to get this merged.

**Comment by joe-niland** [comment]
@pen-pal you may have missed [this](https://github.com/cloudposse/terraform-aws-elasticache-redis/pull/170#discussion_r936247951)

**Review by joe-niland — APPROVED** [review]

**Comment by joe-niland** [comment]
Thanks @pen-pal - released as https://github.com/cloudposse/terraform-aws-elasticache-redis/releases/tag/0.45.0

**Pull request merged**